### ISSUE TITLE
Current version of rabit is incompatible with current version of xgboost

### ISSUE THREAD

**Issue by KOLANICH** [dmlc/xgboost#4274]

**Comment by hcho3** [comment]
@KOLANICH I just tried building XGBoost with the latest dmlc-core and it just builds fine. In what sense is dmlc-core "incompatible" with XGBoost?

**Comment by KOLANICH** [comment]
Yesterdey I have tried to build xgboost (linux, amd64, gcc 8) updating dmlc-core to `master` and got build errors in https://github.com/dmlc/dmlc-core/blob/master/include/dmlc/logging.h#L20 and some other errors related to macros meant to be in config, which are not there. It seems like dmlc-core config is not generated when building XGBoost.

**Comment by hcho3** [comment]
@KOLANICH Can you create a fresh new git clone of XGBoost, upgrade dmlc-core and try building? That’s what I did and I don’t see build failure

**Comment by KOLANICH** [comment]
I have retried. When I update dmlc-core alone, it builds. When I also update rabit, it stops (though the error points into dmlc-core). When I also downgrade dmlc-core, keeping rabit upgraded, it still fails.

**Comment by CodingCat** [comment]
@chenqin can you check this?

**Comment by chenqin** [comment]
Yes, I can reproduce build failure, feel free to assign it to me. It might has something to do with the fact that rabit copied dmlc-core headers and build were failing due to different implied dmlc-core header directory layout.

```
Makefile:31: MAKE [make] - checked OK
g++ -DDMLC_LOG_CUSTOMIZE=1 -std=c++11 -Wall -Wno-unknown-pragmas -Iinclude   -Idmlc-core/include -Irabit/include -I/include -O3 -funroll-loops -msse2 -fPIC -fopenmp -MM -MT build/logging.o src/logging.cc >build/logging.d
In file included from dmlc-core/include/dmlc/./json.h:24:0,
                 from dmlc-core/include/dmlc/parameter.h:25,
                 from src/logging.cc:8:
dmlc-core/include/dmlc/././logging.h:20:10: error: #include expects "FILENAME" or <FILENAME>
 #include DMLC_EXECINFO_H
          ^
Makefile:141: recipe for target 'build/logging.o' failed
make: *** [build/logging.o] Error 1

```

**Comment by hcho3** [comment]
Addressed in dmlc/rabit#86
